sdfObject/sdfthing-oma-device.sdf.json

```diff
@@ -0,0 +1,198 @@
+{
+    "info": {
+      "title": "OMA LwM2M Device (Object ID 3)",
+      "version": "2020-11-10",
+      "copyright": "Copyright 2020 Open Mobile Alliance",
+      "license": "BSD-3-Clause"
+    },
+    "namespace": {
+      "oma": "https://onedm.org/ecosystem/oma"
+    },
+    "defaultNamespace": "oma",
+    "sdfThing": {
+      "Device": {
+        "label": "Device",
+        "description": "This LwM2M Object provides a range of device related information which can be queried by the LwM2M Server, and a device reboot and factory reset function.",
+        "sdfRequired": [
+            "#/sdfObject/Device/sdfAction/",
```

Comment:
```suggestion
            "#/sdfObject/Device/sdfAction/Reboot",
```

Reply:
Indeed.

It seems there is bug in the LwM2M to SDF converter at http://wishi.nomadiclab.com/sdf-converter/. If you try the following XML definition, the sdfRequired contains "#/sdfObject/Test/sdfAction/" instead of "#/sdfObject/Test/sdfAction/MandatoryAction".

```xml
<?xml version="1.0" encoding="UTF-8"?>

<LWM2M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.openmobilealliance.org/tech/profiles/LWM2M.xsd">
  <Object ObjectType="MODefinition">
		<Name>Test</Name>
		<Description1><![CDATA[Test.]]></Description1>
		<ObjectID>777</ObjectID>
		<ObjectURN>urn:oma:lwm2m:oma:777</ObjectURN>
	  	<LWM2MVersion>1.0</LWM2MVersion>
	  	<ObjectVersion>1.0</ObjectVersion>
		<MultipleInstances>Single</MultipleInstances>
		<Mandatory>Optional</Mandatory>
		<Resources>
			<Item ID="0">
				<Name>MandatoryProp</Name>
        		<Operations>R</Operations>
        		<MultipleInstances>Single</MultipleInstances>
				<Mandatory>Mandatory</Mandatory>
				<Type>String</Type>
				<RangeEnumeration></RangeEnumeration>
				<Units></Units>
				<Description><![CDATA[]]></Description>
			</Item>
			<Item ID="1">
				<Name>OptionalProp</Name>
        		<Operations>R</Operations>
        		<MultipleInstances>Single</MultipleInstances>
				<Mandatory>Optional</Mandatory>
				<Type>String</Type>
				<RangeEnumeration></RangeEnumeration>
				<Units></Units>
				<Description><![CDATA[]]></Description>
			</Item>
			<Item ID="2">
				<Name>MandatoryAction</Name>
        		<Operations>E</Operations>
        		<MultipleInstances>Single</MultipleInstances>
				<Mandatory>Mandatory</Mandatory>
				<Type></Type>
				<RangeEnumeration></RangeEnumeration>
				<Units></Units>
				<Description><![CDATA[]]></Description>
			</Item>
			<Item ID="3">
				<Name>OptionalAction</Name>
        		<Operations>E</Operations>
        		<MultipleInstances>Single</MultipleInstances>
				<Mandatory>Optional</Mandatory>
				<Type></Type>
				<RangeEnumeration></RangeEnumeration>
				<Units></Units>
				<Description><![CDATA[]]></Description>
			</Item>
		</Resources>
		<Description2></Description2>
	</Object>
</LWM2M>
```

Reply:
Good catch! [Fixed](https://github.com/EricssonResearch/ipso-odm/commit/e00c8d0e27418e74a99bac4d4cd081c35db965d1) now in the [ipso-odm toolkit](https://github.com/EricssonResearch/ipso-odm). Unfortunately updating the web tool takes a bit more effort and will have to wait for later.

Reply:
Thanks

Reply:
I updated the web tool today, so it should work now.